url5583f3dbb2_XOR19_XScript_tgz-pJ8-xscript_interactive_MainJ8/src/xscript/compiler/inst/XInstConst.java

```diff
@@ -0,0 +1,174 @@
+package xscript.compiler.inst;
+
+import xscript.XOpcode;
+import xscript.compiler.XDataOutput;
+
+public class XInstConst extends XInst {
+
+	private Object _const;
+	
+	public XInstConst(int line, Object _const) {
+		super(line, getOP(_const));
+		this._const = _const;
+	}
+
+	private static XOpcode getOP(Object _const){
+		if(_const==null){
+			return XOpcode.LOADN;
+		}else if(_const instanceof Boolean){
+			if(((Boolean)_const).booleanValue()){
+				return XOpcode.LOAD_TRUE;
+			}
+			return XOpcode.LOAD_FALSE;
+		}else if(_const instanceof Byte){
+			int num = ((Byte)_const).byteValue();
+			switch(num){
+			case -1:
+				return XOpcode.LOADI_M1;
+			case 0:
+				return XOpcode.LOADI_0;
+			case 1:
+				return XOpcode.LOADI_1;
+			case 2:
+				return XOpcode.LOADI_2;
+			}
+			return XOpcode.LOADB;
+		}else if(_const instanceof Byte||_const instanceof Short||_const instanceof Character){
+			int num = 0;
+			if(_const instanceof Character){
+				num = ((Character)_const).charValue();
+			}else{
+				num = ((Number)_const).intValue();
+			}
+			switch(num){
+			case -1:
+				return XOpcode.LOADI_M1;
+			case 0:
+				return XOpcode.LOADI_0;
+			case 1:
+				return XOpcode.LOADI_1;
+			case 2:
+				return XOpcode.LOADI_2;
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 43:52, 59:68.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.